src/OrbitGl/GraphTrack.cpp

```diff
@@ -171,24 +172,39 @@ template <size_t Dimension>
 void GraphTrack<Dimension>::DrawLabel(Batcher& batcher, TextRenderer& text_renderer,
                                       Vec2 target_pos, const std::string& text,
                                       const Color& text_color, const Color& font_color, float z) {
-  uint32_t font_size = layout_->CalculateZoomedFontSize();
+  uint32_t font_size = GetLegendFontSize();
+  const float kTextLeftMargin = 2.f;
+  const float kTextRightMargin = kTextLeftMargin;
+  const float kTextTopMargin = layout_->GetTextOffset();
+  const float kTextBottomMargin = kTextTopMargin;
+  const float kSpaceBetweenLines = layout_->GetTextOffset();
 
   std::vector<std::string> lines = absl::StrSplit(text, '\n');
   float text_width = 0;
   for (const std::string& line : lines) {
     text_width = std::max(text_width, text_renderer.GetStringWidth(line.c_str(), font_size));
   }
-  float text_height = layout_->GetTextBoxHeight() * lines.size();
+  float single_line_height = text_renderer.GetStringHeight(text.c_str(), font_size);
+  float text_height = single_line_height * lines.size() + kSpaceBetweenLines * (lines.size() - 1);
   Vec2 text_box_size(text_width, text_height);
 
   float arrow_width = text_box_size[1] / 2.f;
+  Vec2 arrow_box_size(text_box_size[0] + kTextLeftMargin + kTextRightMargin,
+                      text_box_size[1] + kTextTopMargin + kTextBottomMargin);
   bool arrow_is_left_directed =
-      target_pos[0] < viewport_->GetWorldTopLeft()[0] + text_box_size[0] + arrow_width;
+      target_pos[0] < viewport_->GetWorldTopLeft()[0] + arrow_box_size[0] + arrow_width;
   Vec2 text_box_position(
-      target_pos[0] + (arrow_is_left_directed ? arrow_width : -arrow_width - text_box_size[0]),
+      target_pos[0] + (arrow_is_left_directed ? arrow_width + kTextLeftMargin
+                                              : -arrow_width - kTextRightMargin - text_box_size[0]),
       target_pos[1] - text_box_size[1] / 2.f);
 
-  Box arrow_text_box(text_box_position, text_box_size, z);
+  float bottom_y_of_top_line = text_box_position[1] + text_box_size[1] - single_line_height;
+  text_renderer.AddText(text.c_str(), text_box_position[0], bottom_y_of_top_line, z, text_color,
```

Comment:
So, when writing a multi-line string we have to provide the bottom_y of the top_line? So strange!

Reply:
I guess we only need to specify the location to add the top line in order to add a multiline string. And for the opengl coordinate system, we need to specify the position of the bottom left point.